src/client/sandbox/node/element.js

```diff
@@ -577,14 +578,20 @@ export default class ElementSandbox extends SandboxBase {
 
         if (ElementSandbox._hasShadowUIParentOrContainsShadowUIClassPostfix(el))
             ShadowUI.markElementAndChildrenAsShadow(el);
+
+        if (el.tagName && !domUtils.isShadowUIElement(el))
+            this.liveNodeListFactory.onElementMutation(el);
     }
 
-    onElementRemoved (el) {
+    _onElementRemoved (el) {
         if (domUtils.isBodyElement(el))
             this.shadowUI.onBodyElementMutation();
 
         else if (domUtils.isBaseElement(el))
             urlResolver.updateBase(getDestLocation(), this.document);
+
+        if (el.tagName && !domUtils.isShadowUIElement(el))
+            this.liveNodeListFactory.onElementMutation(el);
```

Comment:
`onElementMutation` looks like bad name.
`onElementAddedOrRemoved` is more cleaner.